libs/payments/ui/src/lib/nestapp/app.ts

```diff
@@ -20,9 +23,17 @@ class AppSingleton {
     return this.app;
   }
 
+  async getLocalizerServer() {
+    return (await this.getApp()).get(LocalizerServer);
+  }
+
   async getCartService() {
     return (await this.getApp()).get(CartService);
   }
 }
 
-export const app = new AppSingleton();
+if (!(global as any)[NEST_APP]) {
+  (global as any)[NEST_APP] = new AppSingleton();
+}
+
+export const app = (global as any)[NEST_APP] as AppSingleton;
```

Comment:
I noticed that the app was being instantiated on every request, so I figured the fix would be to throw the instance onto Global. Got the inspiration for this from [here](https://github.com/vercel/next.js/discussions/15054#discussioncomment-658138).

Reply:
Spoke to @bbangert about this one, and updated implementation to make use of `libs/payments/ui/src/lib/utils/singleton`